### PR TITLE
Bluetooth: Mesh: Add a note about end of transition publication

### PR DESCRIPTION
**Pull request by PavelVPV**
According to MshPRFv1.0.1, section 3.7.6.1.2, when publishing is enabled
for a model, unless specified otherwise, by a higher layer
specification, a corresponding status message shall be published
immediately after the state transition ends.

This behaviour is not implemented in the stack. The models only publish
at the start of transition. An application is responsible for publishing
a status message at the end of transition.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

### CODE REVIEW

File: include/bluetooth/mesh/gen_lvl_srv.h [hunk 1]
```diff
@@ -64,6 +64,12 @@ struct bt_mesh_lvl_srv_handlers {
 			  struct bt_mesh_lvl_status *rsp);
 
 	/** @brief Set the level state.
+	 *
+	 * When a set message is received, the model publishes a status message, with the response
+	 * set to @c rsp. When an acknowledged set message is received, the model also sends a
+	 * response back to a client. If a change of the state is not instantaneous, e.g. when
+	 * @ref bt_mesh_model_transition_time returns a nonzero value, the application is
+	 * responsible for publishing a value of the level state at the end of transition.
```
Comment (alxelax) [inline include/bluetooth/mesh/gen_lvl_srv.h:72]: Other states you wrote from the capital letter but not the level

**Comment by alxelax** [comment]
Are you going to add a description for the scheduler model? It performs publishing on behalf of the Scene and the OnOff models here: https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/bluetooth/mesh/scheduler_srv.c#L474
and here: https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/bluetooth/mesh/scheduler_srv.c#L492

File: include/bluetooth/mesh/gen_lvl_srv.h (hunk 1 above)
Comment (PavelVPV) [reply include/bluetooth/mesh/gen_lvl_srv.h:72]: I followed the pattern in this file. Level is not capitalized here.

**Comment by PavelVPV** [comment]
> Are you going to add a description for the scheduler model? It performs publishing on behalf of the Scene and the OnOff models here: https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/bluetooth/mesh/scheduler_srv.c#L474 and here: https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/bluetooth/mesh/scheduler_srv.c#L492

Is this needed? It will be handled implicitly by an application as if it is a state change caused by a set message.

**Comment by PavelVPV** [comment]
> Could this go into http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/libraries/bluetooth_services/mesh/model_types.html instead?

Thanks for the suggestion, I've added a section about transitions to that page. But I would like to keep the added paragraph in the callbacks' descriptions, because it is the place where a callback behavior has to be described.

**Review by alxelax — APPROVED** [review]

**Review by trond-snekvik — APPROVED** [review]
@mia-ko could you please review the rst changes as well as the API paragraphs?
LGTM.

File: include/bluetooth/mesh/gen_lvl_srv.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (mia-ko) [reply include/bluetooth/mesh/gen_lvl_srv.h:72]: And ideally all header files should be aligned, writing the state names in the same way. But this is left to your own discretion as developers.

**Review by mia-ko — CHANGES_REQUESTED** [review]

**Review by mia-ko — APPROVED** [review]

**Pull request merged**